Packs/CommonScripts/Scripts/IsDomainInternal/IsDomainInternal.py

```diff
@@ -0,0 +1,89 @@
+from time import sleep
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+POLLING_TIME = 2  # Interval to wait in seconds when polling to check if indicator was created
+
+def is_domain_internal(domain, internal_domains):
+    parts = domain.split(".")
+    for i in range(len(parts), 0, -1):
+        sub = ".".join(parts[-i:])
+        if sub in internal_domains:
+            return True
+    return False
+
+def main():
+    internal_domains_list_name = demisto.args().get("InternalDomainsListName", "InternalDomains")
+    domains_to_check = argToList(demisto.args().get("Domains", ()))
+
+    # Get the list of internal domains from the XSOAR list:
+    internal_domains = demisto.executeCommand("getList", {"listName": internal_domains_list_name})[0]['Contents']
+    if "Item not found" in internal_domains:
+        return_error(f"The list name {internal_domains_list_name} does not exist.")
+
+    # Split internal domains from XSOAR list to be a list:
+    if internal_domains:
+        try:
+            internal_domains = internal_domains.split("\n")
+        except Exception as ex:
+            return_error(f"Could not parse the internal domains list. Please make sure that the list contains domain names, separated by new lines.\nThe exact error is: {ex}")
+    else:
+        return_results(f"No internal domains were found under {internal_domains_list_name}.")
+        return
+
+
+    # Create list of domain names with internal property
+    domain_list = [{"Name": domain, "Internal": is_domain_internal(domain, internal_domains)} for domain in domains_to_check]
+
+
+    for domain in domain_list:
+        args_exists_check = {
+            "indicator": domain.get("Name")
+        }
+
+        args_create_or_set_indicator = {
+            "value": domain.get("Name"),
+            "type": "Domain",
+            "internal": domain.get("Internal")
+        }
+
+        # Check if indicator exists:
+        is_exist_res = demisto.executeCommand("CheckIndicatorValue", args_exists_check)
+        indicator_exists = is_exist_res[0]["Contents"][0]["Exists"]
+
+        # If indicator doesn't exist, create it and continuously poll for its creation (which happens asynchronously):
+        if not indicator_exists:
+            demisto.executeCommand("createNewIndicator", args_create_or_set_indicator)
+            while not indicator_exists:  # Looping because it takes time for the indicator to be created
+                is_exist_res = demisto.executeCommand("CheckIndicatorValue", args_exists_check)
+                indicator_exists = is_exist_res[0]["Contents"][0]["Exists"]
+                sleep(POLLING_TIME)
```

Comment:
Please try to not use sleep in the code. Is it necessary to check if the indicator was created?, can we create indicators in a block and not one by one?